tools/run_tests/xds_k8s_test_driver/framework/xds_url_map_testcase.py

```diff
@@ -0,0 +1,590 @@
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A test framework built for urlMap related xDS test cases."""
+
+import abc
+import unittest
+import json
+import time
+import inspect
+import threading
+import functools
+from typing import Any, Union, Tuple, Mapping, Iterable, Optional
+
+from framework import xds_flags
+from framework import xds_k8s_flags
+from framework.infrastructure import gcp
+from framework.infrastructure import k8s
+from framework.infrastructure import traffic_director
+from framework.test_app import client_app
+from framework.test_app import server_app
+
+from absl import flags
+from absl import logging
+from absl.testing import absltest
+from google.protobuf import json_format
+import googleapiclient
+from kubernetes.utils.create_from_yaml import FailToCreateError
+
+logging.set_verbosity(logging.DEBUG)
```

Comment:
I think this should be on the caller to set this, e.g `--verbosity=1`, or more selectively `--logger_levels=__main__:DEBUG,framework:DEBUG`